src/index.ts

```diff
@@ -5,7 +5,12 @@ import {addCommand, rmCommand} from './commands/add-rm';
 import {command as buildCommand} from './commands/build';
 import {command as devCommand} from './commands/dev';
 import {command as installCommand} from './commands/install';
-import {CLIFlags, loadAndValidateConfig} from './config.js';
+import {
+  CLIFlags,
+  loadAndValidateConfig,
+  SnowpackConfig,
+  SnowpackPlugin as SnowpackPluginInit,
```

Comment:
Would it make sense to rename this internally as `SnowpackPluginInit` or `SnowpackPluginConfig`  instead?

Reply:
Relevant convo in #607 : I'd like to continue to use "Snowpack Plugin" to refer to the object, and not the function that returns the object. I believe we were already doing this, but it's also sound for a few reasons: 

- The plugin is the object that we reason about in the rest of the codebase, and the function is just a factory to create them that is then discarded.
- Whenever we talk about how plugins work, we want to refer to them as plugins and not "the object that was returned by a plugin"
- Documentation naming !== code naming leads to trouble long term with communicating with users (when they said plugin, did they mean the object or the function).

Reply:
Happy to call the function `SnowpackPluginFactory`, `SnowpackPluginCreator`, `SnowpackPluginFunction`, etc.

Reply:
> * The plugin is the object that we reason about in the rest of the codebase, and the function is just a factory to create them that is then discarded.
> * Whenever we talk about how plugins work, we want to refer to them as plugins and not "the object that was returned by a plugin"
> * Documentation naming !== code naming leads to trouble long term with communicating with users (when they said plugin, did they mean the object or the function).

All good points. I think maybe `SnowpackPluginFactory` would be a good name for the function.